### PR TITLE
feat(nx): add new nx list command which enables simpler schematic discovery

### PR DESCRIPTION
**Pull request by catfireparty**
e.g.

image

### CODE REVIEW

File: e2e/list.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      },
+      testTimeout
+    );
+  });
+});
+
+forEachCli('angular', () => {
+  describe('list', () => {
+    it(
+      `shouldn't do anything at all`,
```
Comment (vsavkin) [inline e2e/list.test.ts]: Since the implementaiton isn't in tao but in workspace, the command should work with both the Tao and the Angular CLIs, right?

**Review by vsavkin — CHANGES_REQUESTED** [review]

File: e2e/list.test.ts (hunk 1 above)
Comment (jdpearce) [reply e2e/list.test.ts]: How? It's like `dep-graph` or similar in that it only exists on `nx`...

I thought `tao` only contained commands that also exist in the Angular CLI?

File: e2e/list.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+forEachCli('nx', () => {
+  describe('list', () => {
+    beforeEach(() => {
+      ensureProject();
+    });
+
+    it(
+      `should list available collections`,
+      async () => {
+        const listOutput = runCLI('list');
```
Comment (vsavkin) [inline e2e/list.test.ts]: runCommand(
        'npm run nx list"'
      );

File: e2e/list.test.ts (hunk 1 above)
Comment (jdpearce) [reply e2e/list.test.ts]: Changed as per VC last night. I did consider renaming `forEachCli` to something like `forEachWorkspace` but thought that might be a bit much as an incidental change...

**Comment by jdpearce** [comment]
I've added some documentation to this now.

Happy to add more if you think there's somewhere specific I should be highlighting this.

**Comment by vsavkin** [comment]
Looks good

**Pull request merged**